packages/components/src/components/rating-stars/rating-stars.tsx

```diff
@@ -9,298 +9,195 @@
-    this.isHovering = false;
-    this.setValue(this.hoverValue);
-
-    // Prevent click on mobile devices
-    event.preventDefault();
-  }
-
-  setValue(newValue: number) {
-    if (this.disabled) {
-      return;
-    }
+    this.scaleChange.emit({ value: this.rating });
+  };
 
-    this.value = newValue === this.value ? 0 : newValue;
-    this.emitRatingValue();
-    this.isHovering = false;
-  }
+  renderStar(index: number, selected = false, rating: number) {
+    const size = sizes[this.starSize];
+    const isWholeNumber = rating % 1 === 0;
+    const isLastNumber = Math.ceil(rating) === index;
 
-  getValueFromXPosition(evTou?: TouchEvent, evKey?: MouseEvent) {
-    const positionX = evTou ? evTou.touches[0].clientX : evKey.clientX;
-    const containerLeft = this.element.getBoundingClientRect().left;
-    const containerWidth = this.element.getBoundingClientRect().width;
-    return clamp(
-      this.roundToPrecision(
-        ((positionX - containerLeft) / containerWidth) * this.max,
-        this.precision
-      ),
-      0,
-      this.max
+    return (
+      <div
+        part="star"
+        data-value={index}
+        data-selected={selected}
+        data-half={isLastNumber && !isWholeNumber}
+        onMouseUp={!this.readonly && this.handleStarClick}
+        onTouchEnd={!this.readonly && this.handleTouchEnd}
+      >
+        <scale-icon-action-favorite size={size} part="placeholder-star" />
+        <div class="icon-clip">
+          <scale-icon-action-favorite
+            size={size}
+            selected
+            part="selected-star"
+          />
+        </div>
+      </div>
     );
   }
 
-  roundToPrecision(numberToRound: number, precision = 1) {
-    const multiplier = 1 / precision;
-    return Math.ceil(numberToRound * multiplier) / multiplier;
-  }
+  renderRating() {
+    const stars = [];
+    const roundedRating = Math.ceil(this.rating);
+    const min = this.minRating === 0 ? this.minRating + 1 : this.minRating;
+    const max = this.maxRating;
 
-  getAriaLabel() {
-    return this.ariaLabelTranslation
-      .replace(/\$\{value\}/gi, this.value.toString())
-      .replace(/\$\{max\}/gi, this.max.toString());
-  }
+    for (let index = min; index <= max; index++) {
+      const isSelected = roundedRating >= index;
+      stars.push(this.renderStar(index, isSelected, this.rating));
+    }
 
-  emitRatingValue() {
-    this.scaleRatingChange.emit({ value: this.value });
+    return stars;
   }
 
   render() {
-    const counter = Array.from(Array(this.max).keys());
-    const displayValue = this.isHovering ? this.hoverValue : this.value;
-
     return (
       <Host>
+        {this.label && (
+          <label
+            id={`${this.ratingStarId}-label`}
+            part="label"
+            htmlFor={this.ratingStarId}
+          >
+            {this.label}
+          </label>
+        )}
+        <input
+          disabled={this.disabled || this.readonly}
+          part="range-slider"
```

Comment:
keyboard support is flaky, when you reach the component by tabbing, keyboard control works (on MS Edge), but after I click, then it doesn't. I'm thinking that maybe the focus target is different in each case… could that be?

Reply:
I have changed this behaviour in a way that on clicking a star the input gets the focus. Now, when clicking a star the arrow keys are still functional.

Reply:
I wasn't able to check this but sounds great, if it comes back (hopefully not), we'll notice 🤓